From 8ca990c6f3a7cdff08543329fc81e23ea6ba553d Mon Sep 17 00:00:00 2001
From: Christoffer Sandberg <cs@tuxedo.de>
Date: Mon, 12 May 2025 11:51:54 +0200
Subject: [PATCH 3/4] drm/i915/rc6: Disable RC6 for TUXEDO InfinityBook Pro
 Gen8 Intel

Disable RC6 for the TUXEDO InfinityBook Pro Gen8 Intel to prevent random
rare hard resets.

Signed-off-by: Christoffer Sandberg <cs@tuxedo.de>
Signed-off-by: Werner Sembach <wse@tuxedocomputers.com>
---
 drivers/gpu/drm/i915/gt/intel_rc6.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/drm/i915/gt/intel_rc6.c b/drivers/gpu/drm/i915/gt/intel_rc6.c
index e91e5cdca26ca..1e17a99231203 100644
--- a/drivers/gpu/drm/i915/gt/intel_rc6.c
+++ b/drivers/gpu/drm/i915/gt/intel_rc6.c
@@ -3,6 +3,7 @@
  * Copyright © 2019 Intel Corporation
  */
 
+#include <linux/dmi.h>
 #include <linux/pm_runtime.h>
 #include <linux/string_helpers.h>
 
@@ -496,6 +497,11 @@ static bool rc6_supported(struct intel_rc6 *rc6)
 	struct drm_i915_private *i915 = rc6_to_i915(rc6);
 	struct intel_gt *gt = rc6_to_gt(rc6);
 
+	if (dmi_match(DMI_BOARD_NAME, "PH4PRX1_PH6PRX1") ||
+	    dmi_match(DMI_BOARD_NAME, "PH4PG31") ||
+	    dmi_match(DMI_BOARD_NAME, "PH6PG01_PH6PG71"))
+		return false;
+
 	if (!HAS_RC6(i915))
 		return false;
 
-- 
2.43.0

